### PR TITLE
document kubernetes readiness pitfall

### PR DESCRIPTION
**Pull request by StarpTech**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [X] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — COMMENTED** [review]
Can this not be added to the recommendations document?

**Comment by StarpTech** [comment]
Good idea.

File: docs/Recommendations.md [hunk 1]
```diff
@@ -161,3 +161,23 @@ backend static-backend
 [scale-horiz]: https://en.wikipedia.org/wiki/Scalability#Horizontal
 [why-use]: https://web.archive.org/web/20190821102906/https://medium.com/intrinsic/why-should-i-use-a-reverse-proxy-if-node-js-is-production-ready-5a079408b2ca
 [haproxy]: https://www.haproxy.org/
+
+## Kubernetes
```
Comment (jsumners) [inline docs/Recommendations.md:165]: The TOC needs to be updated on line 8.

File: docs/Recommendations.md [hunk 2]
```diff
@@ -161,3 +161,23 @@ backend static-backend
...
+        path: /health
+        port: 4000
+    initialDelaySeconds: 30
+    periodSeconds: 30
+    timeoutSeconds: 3
+    successThreshold: 1
+    failureThreshold: 5
+```
+
+The `readinessProbe` use [(by default](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes)) the pod IP as hostname. Fastify listen on `127.0.0.1` by default. Your probe won't be able to reach your application. In order to make it work you have to listen on `0.0.0.0` or specify a custom hostname in the `readinessProbe.httpGet` spec.
```
Comment (jsumners) [inline docs/Recommendations.md:183]:
```suggestion
The `readinessProbe` uses [(by default](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes)) the pod IP as the hostname. Fastify listens on `127.0.0.1` by default. The probe will not be able to reach the application in this case. In order to make it work, the application must listen on `0.0.0.0` or specify a custom hostname in the `readinessProbe.httpGet` spec, as per the following example:

```yaml
readinessProbe:
    httpGet:
        path: /health
        port: 4000
    initialDelaySeconds: 30
    periodSeconds: 30
    timeoutSeconds: 3
    successThreshold: 1
    failureThreshold: 5
```
```

**Comment by zekth** [comment]
Might be good to mention that the route needs to be created; wrote like this we can assume that the `health` route is generated by default.

**Review by jsumners — APPROVED** [review]
LGTM

**Comment by StarpTech** [comment]
@zekth I think it's natural that you need an endpoint if you want to provide a probe URL. :smile:

File: docs/Recommendations.md [hunk 3]
```diff
@@ -161,3 +162,18 @@ backend static-backend
 [scale-horiz]: https://en.wikipedia.org/wiki/Scalability#Horizontal
 [why-use]: https://web.archive.org/web/20190821102906/https://medium.com/intrinsic/why-should-i-use-a-reverse-proxy-if-node-js-is-production-ready-5a079408b2ca
 [haproxy]: https://www.haproxy.org/
+
+## Kubernetes
```
Comment (jsumners) [inline docs/Recommendations.md:166]:
```suggestion
## Kubernetes
<a id="kubernetes"></a>
```

**Comment by zekth** [comment]
> @zekth I think it's natural that you need an endpoint if you want to provide a probe URL. 😄

Sure but mentionned this way we can think that fastify creates a `health` route by default. That's just my point.

**Comment by StarpTech** [comment]
I got your point but I'd like to keep it concise.

**Pull request merged**